### ISSUE TITLE
[v2] Telepresence leave can't find an intercept after sleep mode

### ISSUE THREAD

**Issue by andrewnazarov** [telepresenceio/telepresence#2244]
For a successful intercept after the sleep mode `telepresence leave` returns `telepresence: error: Intercept named "my_service" not found` and I can't stop intercepting with this command.

**To Reproduce**
1. Run `telepresence intercept notifications --port 8080`
2. I see that the interception is created:
```
> telepresence list
...
notifications                       : intercepted
    Intercept name: notifications
    State         : ACTIVE
    Workload kind : Deployment
    Destination   : 127.0.0.1:8080
    Intercepting  : all TCP connections
...
```
3. I go to the sleep mode for a while
4. Then I run `telepresence leave notifications` to stop intercepting
5. And I get `telepresence: error: Intercept named "notifications" not found`

**Expected behavior**
Intercepting is stopped

**Versions (please complete the following information):**
telepresence version
Client: v2.4.7 (api v3)
Root Daemon: v2.4.7 (api v3)
User Daemon: v2.4.7 (api v3)

MacOS Mojave 10.14.5
GKE 1.21.5-gke.1802

**Additional context**
I don't have a running service locally, however, previously it didn't matter and the aforementioned command worked. It might be the case that my laptop went into sleep mode for a while.

**Comment by andrewnazarov** [comment]
It seems the interception stopped working because my calls hit the real service that is deployed to a K8s cluster. However, `telepresence list` still shows that the interception is `ACTIVE` which is wrong.

Also, I can see the following message in a traffic-agent container:
```
2021-11-30 10:17:11.5400 info    client : The previously-active intercept has been deleted
```
This is more likely a time my laptop went into sleep mode.

**Comment by andrewnazarov** [comment]
Ok, it seems it caches some information or cannot update it.

Also, I noticed that after the situation that I just described I can't do any other intercepts - it gets stuck and the client failed with a timeout later on. And `telepresence list` keeps showing the old info that traffic-agent is not installed even though it is installed correctly according to traffic-agent logs.

I managed to fix the problem by doing `telepresence quit`. After that `telepresence list` started showing the correct info about existing traffic-agents.

**Comment by andrewnazarov** [comment]
I modified the initial message.

**Comment by andrewnazarov** [comment]
I can't reproduce the error anymore. Sleep mode or a network outage up to an hour lead to all Internet addresses being unreachable until I do `telepresence quit`. Probably it should be an additional issue for this behaviour.

**Comment by richardscarrott** [comment]
I've experienced the same issue on a Macbook Pro M1 a few times. I come back to my computer after it's been asleep and service discovery starts to fail so I have to re-intercept but I'm unable to leave the existing one without quitting.

```
Client: v2.4.10 (api v3)
Root Daemon: v2.4.10 (api v3)
User Daemon: v2.4.10 (api v3)
```

```
> telepresence list

api-gateway  : intercepted
    Intercept name    : api-gateway
    State             : ACTIVE
    Workload kind     : Deployment
    Destination       : 127.0.0.1:6060
    Volume Mount Point: /var/folders/sm/nk122_b57j988xsjbg1szyvh0000gn/T/telfs-1725349927
    Intercepting      : all TCP connections
users        : ready to intercept (traffic-agent not yet installed)
...

> telepresence leave api-gateway
telepresence: error: Intercept named "api-gateway" not found
```

Not sure if it's just me, and I know this isn't a particularly useful comment, but having used telepresence 1 quite successfully for a few years, I've found telepresence 2 to get itself into all sorts of broken states which it can't recover from; often I have to manually restart telepresence, redeploy k8s deployments as well as kill any unused docker containers which haven't been cleaned up. I'm trying to compile the various issues so I can document them better, not sure if there's a specific cmd to dump out useful debugging information when I hit these problems?

**Comment by tanmaymishu** [comment]
I have the exact same issue:

```
$ telepresence list
auth-depl      : intercepted
    Intercept name         : auth-depl
    State                  : ACTIVE
    Workload kind          : Deployment
    Destination            : 127.0.0.1:3000
    Service Port Identifier: 3000
    Intercepting           : all TCP connections
```

```
$ telepresence leave auth-depl
telepresence: error: Intercept named "auth-depl" not found
```

**Comment by thallgren** [comment]
@tanmaymishu two questions.
1. If you repeat `telepresence list` after you get that error, does it still show state `ACTIVE`?
2. Can you show the output of `kubectl logs -f <intercepted pod> traffic-agent`?

**Comment by thallgren** [comment]
@richardscarrott regarding dumping useful debug information. Try:
1. `telepresence loglevel debug`
2. Reproduce the problem
3. `telepresence gather-logs`

**Comment by tanmaymishu** [comment]
@thallgren, yes, it does still show the ACTIVE state. This is the second time I have encountered the issue. Now on my M1 chip and a few hours ago on an Intel chip.

The output (I have redacted some keys and ids):

```
2022-02-14 14:54:32.4603 info    Logging at this level "info"
2022-02-14 14:54:32.4604 info    Sidecar version v1.11.8
2022-02-14 14:54:32.4605 info    LoadConfig: processing environment...
2022-02-14 14:54:32.4606 info    supportsHTTP2: probing for HTTP/2 support... : base-url="http://127.0.0.1:3000"
2022-02-14 14:54:33.9640 info    supportsHTTP2: result =  false : base-url="http://127.0.0.1:3000"
2022-02-14 14:54:33.9644 info    LoadConfig: processing annotations...
2022-02-14 14:54:33.9646 info    LoadConfig: asking traffic-manager for license information...
2022-02-14 14:54:33.9819 error   error(s) getting license: error reading license: open /home/telepresence/license: no such file or directory
error reading hostDomain: open /home/telepresence/hostDomain: no such file or directory

2022-02-14 14:54:33.9826 info    LoadConfig: asking traffic-manager for Ambassador Cloud configuration...
2022-02-14 14:54:33.9897 info    trying to connect to ambassador cloud at app.getambassador.io:443
2022-02-14 14:54:34.2758 info    connected to ambassador cloud at app.getambassador.io:443
2022-02-14 14:54:34.2776 info    LoadConfig: done
2022-02-14 14:54:34.2781 info    config = {Config:{Name:auth-depl Namespace:default PodIP:10.244.1.70 AgentPort:9900 AppMounts:/tel_app_mounts AppPort:3000 ManagerHost:traffic-manager.ambassador ManagerPort:8081 APIPort:0} EnvoyLogLevel:info EnvoyBaseID:1 DevEnvoyImage: AppProto: UseHTTP2:false TerminatingTLSSecret: OriginatingTLSSecret: License: LicenseHost: LicensePublicKeys:{KeyMap:map[auth.datawire.io:-----BEGIN PUBLIC KEY-----
[REDACTED]
-----END PUBLIC KEY----- beta-auth.datawire.io:-----BEGIN PUBLIC KEY-----
[REDACTED]
-----END PUBLIC KEY-----]} ClusterID:[REDACTED] CanConnectCloud:true SystemAHost:app.getambassador.io SystemAPort:443}
2022-02-14 14:54:34.2852 info    sidecar : new agent secrets mount path: /var/run/secrets/kubernetes.io
2022-02-14 14:54:34.3059 info    sidecar : Connected to Manager 2.4.11
time="2022-02-14T14:54:34Z" level=error msg="V2 Snapshot inconsistency: mismatched route reference and resource lengths: map[] != 1: {\"Resources\":[{\"Version\":\"v1\",\"Items\":{}},{\"Version\":\"v1\",\"Items\":{\"app\":{\"name\":\"app\",\"ClusterDiscoveryType\":{\"Type\":1},\"connect_timeout\":{\"seconds\":10},\"load_assignment\":{\"cluster_name\":\"app\",\"endpoints\":[{\"lb_endpoints\":[{\"HostIdentifier\":{\"Endpoint\":{\"address\":{\"Address\":{\"SocketAddress\":{\"address\":\"127.0.0.1\",\"PortSpecifier\":{\"PortValue\":3000}}}}}}}]}]},\"LbConfig\":null}}},{\"Version\":\"v1\",\"Items\":{\"application_route\":{\"name\":\"application_route\",\"virtual_hosts\":[{\"name\":\"all-the-hosts\",\"domains\":[\"*\"],\"routes\":[{\"match\":{\"PathSpecifier\":{\"Prefix\":\"/\"}},\"Action\":{\"Route\":{\"ClusterSpecifier\":{\"Cluster\":\"app\"},\"HostRewriteSpecifier\":null,\"timeout\":{}}}}]}]}}},{\"Version\":\"v1\",\"Items\":{}},{\"Version\":\"\",\"Items\":null},{\"Version\":\"v1\",\"Items\":{}}]}"
2022/02/14 14:54:44 Memory Usage 0.03Gi
    PID 1, 0.05Gi: /ko-app/app-sidecar agent
    PID 24, 0.05Gi: envoy-static-stripped -l info --bootstrap-version 2 -c bootstrap-ads.yaml --base-id 1
2022-02-14 15:00:21.9189 info    sidecar : Ambassador Cloud has allowed the intercept
2022-02-14 15:00:22.8949 info    sidecar : Ambassador Cloud has allowed the intercept
2022-02-14 15:00:24.2310 info    sidecar : Ambassador Cloud has allowed the intercept
2022-02-14 15:03:14.0087 info    sidecar : Accept got connection from 127.0.0.1:37738 : lis=":0"
2022-02-14 15:04:08.0018 info    sidecar : Accept got connection from 127.0.0.1:37810 : lis=":0"
2022-02-14 15:04:33.2596 info    sidecar : Accept got connection from 127.0.0.1:37838 : lis=":0"
2022-02-14 15:09:35.6625 info    sidecar : Accept got connection from 127.0.0.1:38180 : lis=":0"
2022-02-14 15:14:18.7446 info    sidecar : Accept got connection from 127.0.0.1:38500 : lis=":0"
2022/02/14 15:37:14 Memory Usage 0.04Gi
    PID 1, 0.06Gi: /ko-app/app-sidecar agent
    PID 24, 0.06Gi: envoy-static-stripped -l info --bootstrap-version 2 -c bootstrap-ads.yaml --base-id 1
```

Note: Quitting telepresence with `telepresence quit` and running the `telepresence intercept` again does start to work again. The problem occurs when the Macbook goes to sleep mode.

**Comment by thallgren** [comment]
This problem was due to loss of session information and was fixed by in #2545.
